gobblin-utility/src/main/java/org/apache/gobblin/util/LoggingUncaughtExceptionHandler.java

```diff
@@ -42,6 +45,7 @@ public LoggingUncaughtExceptionHandler(Optional<Logger> logger) {
 
   @Override
   public void uncaughtException(Thread t, Throwable e) {
+    exception = e;
```

Comment:
The exception variable is set but never used, maybe missing something?

Reply:
it's being used in get method for the main thread to determine whether there is an exception in this thread: https://github.com/apache/gobblin/pull/3862/files#diff-5486722712e94a92c9e9129f7b3768c87cd7c3f3cf08613c24b9a5fd4131e6ebR163